odxtools/decodestate.py

```diff
@@ -24,6 +24,12 @@ class DecodeState:
     #: (if not explicitly specified by the object to be decoded.)
     cursor_position: int = 0
 
+    #: the bit position [0, 7] where the object to be extracted begins
+    #:
+    #: If bit position is undefined (`None`), the object to be extracted
+    #: starts at bit 0.
+    cursor_bit_position: Optional[int] = None
```

Comment:
I don't see why this can't be 0 by default

Reply:
the idea is that `None` means "not specified" while 0 indicates that it was explicitly set to 0 and thus code where specifying a bit position is not allowed can error out even if 0 was specified. (with this approach, code where bit positions are allowed needs to use `decode_state.cursor_bit_position or 0`, which is not too bad IMO...)

Reply:
Any case where None and 0 would have different logic?

Reply:
all parameters which do not accept bit positions, e.g. structures, muxes, value parameters with non-integer values, etc...

Reply:
All of those either require bit position to be non specified or 0

Reply:
point taken & code changed. I wanted to be able to check if a bit position is set to 0 where it is not allowed to be defined in the decoding logic, but I finally saw the light and concluded that this is the wrong abstraction level to check this...